src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -0,0 +1,328 @@
+
+#pragma warning disable CA2213
+
+    /// <summary>
+    ///   Pairs containing window reordering nodes and what sibling of theirs is an ancestor of this class.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Used to pass the connections from windows.
+    ///   </para>
+    /// </remarks>
+    private List<(AddWindowReorderingSupportToSiblings Node, Node Sibling)> windowReorderingAncestors = new();
+
+    /// <summary>
+    ///   A sibling that is an ancestor of a window that is currently on top of others.
+    /// </summary>
+    private Node? topSibling;
+
+    private Node parent = null!;
+
+#pragma warning restore CA2213
+
+    private bool isReady;
+
+    /// <summary>
+    ///   Used to save what node to reorder when a window requires to be reordered.
+    /// </summary>
+    private System.Collections.Generic.Dictionary<CustomWindow, Node> connectedWindows = new();
+
+    /// <summary>
+    ///   Used to save what siblings are part of the reordering system to prevent reordering more than is necessary.
+    /// </summary>
+    private HashSet<Node> connectedSiblings = new();
+
+    /// <summary>
+    ///   Finds window reordering nodes in ancestors, looks for manual paths if set, otherwise uses automatic search to
+    ///   find first one.
+    /// </summary>
+    /// <returns>
+    ///   Pairs containing window reordering nodes and what sibling of theirs is an ancestor of this class.
+    /// </returns>
+    public static IEnumerable<(AddWindowReorderingSupportToSiblings Node, Node Sibling)> GetWindowReorderingAncestors(
+        Node startingNode, int maxSearchDepth, Array<NodePath>? paths)
+    {
+        if (paths is null || paths.Count == 0)
+        {
+            foreach (var windowReorderingSupport in GetWindowReorderingAncestors(startingNode, maxSearchDepth))
+            {
+                yield return windowReorderingSupport;
+            }
+        }
+        else
+        {
+            foreach (var windowReorderingSupport in GetWindowReorderingAncestors(startingNode, paths))
+                yield return windowReorderingSupport;
+        }
+    }
+
+    public override void _ExitTree()
+    {
+        windowReorderingAncestors.Clear();
+        connectedWindows.Clear();
+        connectedSiblings.Clear();
+        isReady = false;
+    }
+
+    public void ConnectWindow(CustomWindow window, Node topNode, bool recursive)
+    {
+        // Make sure to be connected to other window reordering nodes before trying to connect the window
+        Setup();
+
+        if (recursive)
+        {
+            // Connection is recursive, connect to other window reordering nodes
+            for (int i = 0; i < windowReorderingAncestors.Count; i++)
+            {
+                windowReorderingAncestors[i].Node.ConnectWindow(
+                    window, windowReorderingAncestors[i].Sibling, recursive);
+            }
+        }
+
+        window.Connect(nameof(Dragged), this, nameof(OnWindowReorder));
```

Comment:
Is there a possibility of accidentally double registering a window? For example if a partial subtree exits the scene tree and is then added back in?

Reply:
From my testing I ran into issues only when I accidentally set manual connections in a wrong way. 
But it might be possible? I will add a check here and in DisconnectWindow to be sure that it won't break.

Reply:
Well if you make a duplicate Connect call, Godot will yell an error at you. So we should really avoid that situation. We have a method for connecting a signal if not connected already, but I think a better approach would be to ensure that if this is called with an already registered window (to this object) an error is thrown / printed and nothing is done.

Reply:
So should it be thrown or just printed? To me printing it sounds enough.

Reply:
I guess printing would be better in this case to not randomly crash if the problem ever occurs.